lightning failed to import parquet file if table column name contains upper case character

## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.
use lightning to load some parquet files from s3. lightning failed with following errors:
```
[2020/11/17 12:27:25.487 +00:00] [INFO] [restore.go:1932] ["restore file start"] [table=`parquet`.`usertable`] [engineNumber=0] [fileIndex=1] [path=uuid-384g0-aurora-parquet/uuid-384g-aurora/test/test.usertable/10/part-00000-1d2efb15-46b7-4e2b-9d9e-81f5209dac0d-c000.gz.parquet:0]
[2020/11/17 12:27:25.630 +00:00] [ERROR] [restore.go:1083] ["encode kv data and write failed"] [table=`parquet`.`usertable`] [engineNumber=0] [takeTime=3.408112523s] [error="unknown columns in header [YCSB_KEY strikerUserCharacterId myStrikerUserCharacterId staminaModifiedAt hotSpotStamina hotSpotStaminaModifiedAt characterCapacity extendedCharacterCapacity drewTutorialGacha isTutorialEnd medalId noun1TitleId conjunctionTitleId noun2TitleId lointAt lastLoginBonusAt]"]
[2020/11/17 12:27:25.630 +00:00] [ERROR] [restore.go:931] ["restore engine failed"] [table=`parquet`.`usertable`] [engineNumber=0] [takeTime=3.408216608s] [error="unknown columns in header [YCSB_KEY strikerUserCharacterId myStrikerUserCharacterId staminaModifiedAt hotSpotStamina hotSpotStaminaModifiedAt characterCapacity extendedCharacterCapacity drewTutorialGacha isTutorialEnd medalId noun1TitleId conjunctionTitleId noun2TitleId lointAt lastLoginBonusAt]"]
[2020/11/17 12:27:25.767 +00:00] [INFO] [restore.go:1932] ["restore file start"] [table=`parquet`.`usertable`] [engineNumber=1] [fileIndex=1] [path=uuid-384g0-aurora-parquet/uuid-384g-aurora/test/test.usertable/2878/part-00000-b8d4a4d2-f56a-4e4c-a48b-f32738272309-c000.gz.parquet:0]
[2020/11/17 12:27:25.884 +00:00] [ERROR] [restore.go:1083] ["encode kv data and write failed"] [table=`parquet`.`usertable`] [engineNumber=1] [takeTime=3.662541533s] [error="unknown columns in header [YCSB_KEY strikerUserCharacterId myStrikerUserCharacterId staminaModifiedAt hotSpotStamina hotSpotStaminaModifiedAt characterCapacity extendedCharacterCapacity drewTutorialGacha isTutorialEnd medalId noun1TitleId conjunctionTitleId noun2TitleId lointAt lastLoginBonusAt]"]
[2020/11/17 12:27:25.884 +00:00] [ERROR] [restore.go:931] ["restore engine failed"] [table=`parquet`.`usertable`] [engineNumber=1] [takeTime=3.662656037s] [error="unknown columns in header [YCSB_KEY strikerUserCharacterId myStrikerUserCharacterId staminaModifiedAt hotSpotStamina hotSpotStaminaModifiedAt characterCapacity extendedCharacterCapacity drewTutorialGacha isTutorialEnd medalId noun1TitleId conjunctionTitleId noun2TitleId lointAt lastLoginBonusAt]"]
[2020/11/17 12:27:25.884 +00:00] [ERROR] [restore.go:953] ["import whole table failed"] [table=`parquet`.`usertable`] [takeTime=3.662745594s] [error="unknown columns in header [YCSB_KEY strikerUserCharacterId myStrikerUserCharacterId staminaModifiedAt hotSpotStamina hotSpotStaminaModifiedAt characterCapacity extendedCharacterCapacity drewTutorialGacha isTutorialEnd medalId noun1TitleId conjunctionTitleId noun2TitleId lointAt lastLoginBonusAt]"]
[2020/11/17 12:27:25.884 +00:00] [ERROR] [restore.go:693] ["restore table failed"] [table=`parquet`.`usertable`] [takeTime=53m47.021028337s] [error="restore table `parquet`.`usertable` failed: unknown columns in header [YCSB_KEY strikerUserCharacterId myStrikerUserCharacterId staminaModifiedAt hotSpotStamina hotSpotStaminaModifiedAt characterCapacity extendedCharacterCapacity drewTutorialGacha isTutorialEnd medalId noun1TitleId conjunctionTitleId noun2TitleId lointAt lastLoginBonusAt]"]
```

--- a/lightning/mydump/parquet_parser.go
+++ b/lightning/mydump/parquet_parser.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"io"
 	"reflect"
+	"strings"
 	"time"
 
 	"github.com/pingcap/br/pkg/storage"
@@ -89,7 +90,7 @@ func NewParquetParser(
 	for i, c := range reader.SchemaHandler.SchemaElements {
 		if c.GetNumChildren() == 0 {
 			// the SchemaElement.Name is capitalized, we should use the original name
-			columns = append(columns, reader.SchemaHandler.Infos[i].ExName)
+			columns = append(columns, strings.ToLower(reader.SchemaHandler.Infos[i].ExName))
 			columnMetas = append(columnMetas, c)
 		}
 	}

--- a/lightning/mydump/parquet_parser.go
+++ b/lightning/mydump/parquet_parser.go
@@ -87,10 +87,11 @@ func NewParquetParser(
 
 	columns := make([]string, 0, len(reader.Footer.Schema)-1)
 	columnMetas := make([]*parquet.SchemaElement, 0, len(reader.Footer.Schema)-1)
-	for i, c := range reader.SchemaHandler.SchemaElements {
+	for _, c := range reader.SchemaHandler.SchemaElements {
 		if c.GetNumChildren() == 0 {
-			// the SchemaElement.Name is capitalized, we should use the original name
-			columns = append(columns, strings.ToLower(reader.SchemaHandler.Infos[i].ExName))
+			// NOTE: the SchemaElement.Name is capitalized, SchemaHandler.Infos.ExName is the raw column name
+			// though in this context, there is no difference between these two fields
+			columns = append(columns, strings.ToLower(c.Name))
 			columnMetas = append(columnMetas, c)
 		}
 	}

--- a/lightning/mydump/parquet_parser_test.go
+++ b/lightning/mydump/parquet_parser_test.go
@@ -19,8 +19,8 @@ var _ = Suite(testParquetParserSuite{})
 
 func (s testParquetParserSuite) TestParquetParser(c *C) {
 	type Test struct {
-		S string `parquet:"name=s, type=UTF8, encoding=PLAIN_DICTIONARY"`
-		A int32  `parquet:"name=a, type=INT32"`
+		S string `parquet:"name=sS, type=UTF8, encoding=PLAIN_DICTIONARY"`
+		A int32  `parquet:"name=a_A, type=INT32"`
 	}
 
 	dir := c.MkDir()
@@ -50,7 +50,7 @@ func (s testParquetParserSuite) TestParquetParser(c *C) {
 	c.Assert(err, IsNil)
 	defer reader.Close()
 
-	c.Assert(reader.Columns(), DeepEquals, []string{"s", "a"})
+	c.Assert(reader.Columns(), DeepEquals, []string{"ss", "a_a"})
 
 	verifyRow := func(i int) {
 		c.Assert(reader.lastRow.RowID, Equals, int64(i+1))

--- a/tests/checkpoint_parquet/parquet.go
+++ b/tests/checkpoint_parquet/parquet.go
@@ -21,7 +21,7 @@ var (
 
 func genParquetFile(dir, name string, count int) error {
 	type Test struct {
-		I int32  `parquet:"name=i, type=INT32"`
+		I int32  `parquet:"name=iVal, type=INT32"`
 		S string `parquet:"name=s, type=UTF8, encoding=PLAIN_DICTIONARY"`
 	}
 

--- a/tests/checkpoint_parquet/run.sh
+++ b/tests/checkpoint_parquet/run.sh
@@ -25,7 +25,8 @@ do_run_lightning() {
 
 mkdir -p $DBPATH
 echo 'CREATE DATABASE cppq_tsr;' > "$DBPATH/cppq_tsr-schema-create.sql"
-echo 'CREATE TABLE tbl(i INT, s VARCHAR(16));' > "$DBPATH/cppq_tsr.tbl-schema.sql"
+# column "iVal" use for testing column name with upper case is properly handled
+echo 'CREATE TABLE tbl(iVal INT, s VARCHAR(16));' > "$DBPATH/cppq_tsr.tbl-schema.sql"
 bin/parquet_gen --dir $DBPATH --schema cppq_tsr --table tbl --chunk 1 --rows $ROW_COUNT
 
 # Set the failpoint to kill the lightning instance as soon as one batch data is written
@@ -38,10 +39,10 @@ run_sql 'DROP DATABASE IF EXISTS checkpoint_test_parquet'
 set +e
 run_lightning -d "$DBPATH" --backend tidb --enable-checkpoint=1 2> /dev/null
 set -e
-run_sql 'SELECT count(*), sum(i) FROM `cppq_tsr`.tbl'
+run_sql 'SELECT count(*), sum(iVal) FROM `cppq_tsr`.tbl'
 check_contains "count(*): 32"
 # sum(0..31)
-check_contains "sum(i): 496"
+check_contains "sum(iVal): 496"
 
 # check chunk offset and update checkpoint current row id to a higher value so that
 # if parse read from start, the generated rows will be different
@@ -53,7 +54,7 @@ set +e
 run_lightning -d "$DBPATH" --backend tidb --enable-checkpoint=1 2> /dev/null
 set -e
 
-run_sql 'SELECT count(*), sum(i) FROM `cppq_tsr`.tbl'
+run_sql 'SELECT count(*), sum(iVal) FROM `cppq_tsr`.tbl'
 check_contains "count(*): 100"
 # sum(0..99)
-check_contains "sum(i): 4950"
+check_contains "sum(iVal): 4950"

